pilot/pkg/model/push_context.go

```diff
@@ -512,6 +512,65 @@ func (ps *PushContext) UpdateMetrics() {
 	}
 }
 
+// GatewayServices returns the set of services which are refered from the proxy gateways.
+func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
```

Comment:
Can we add some tests to this new function to ensure it returns right set of services for various virtual services?

Reply:
+1 this PR has no testing at all and is off by default. It could crash immediately and we wouldn't know (well, I did run it locally - but as part of CI)